homeassistant/components/dlna_dmr/config_flow.py

```diff
@@ -93,133 +95,182 @@ async def async_step_user(self, user_input: FlowInput = None) -> FlowResult:
+        try:
+            await self._async_connect()
+        except ConnectError as err:
+            # This will require user action
+            LOGGER.debug("Entry %s not imported yet: %s", self._location, err.args[0])
+            return await self.async_step_import_turn_on()
+
+        LOGGER.debug("Entry %s ready for import", self._location)
+        return await self._async_create_entry()
+
+    async def async_step_import_turn_on(
+        self, user_input: FlowInput = None
+    ) -> FlowResult:
+        """Request the user to turn on the device so that import can finish."""
+        LOGGER.debug("async_step_import_turn_on: %s", user_input)
+
+        self.context["title_placeholders"] = {"name": self._name or self._location}
 
-        LOGGER.debug("Entry %s ready for import", import_data[CONF_URL])
-        return await self._async_create_entry_from_discovery(discovery, options)
+        errors = {}
+        if user_input is not None:
+            try:
+                await self._async_connect()
+            except ConnectError as err:
+                errors["base"] = err.args[0]
+            else:
+                return await self._async_create_entry()
+
+        self._set_confirm_only()
+        return self.async_show_form(step_id="import_turn_on", errors=errors)
 
     async def async_step_ssdp(self, discovery_info: DiscoveryInfoType) -> FlowResult:
         """Handle a flow initialized by SSDP discovery."""
         LOGGER.debug("async_step_ssdp: discovery_info %s", pformat(discovery_info))
 
-        self._discoveries = [discovery_info]
-
-        udn = discovery_info[ssdp.ATTR_SSDP_UDN]
-        location = discovery_info[ssdp.ATTR_SSDP_LOCATION]
+        self._location = discovery_info[ssdp.ATTR_SSDP_LOCATION]
+        self._udn = discovery_info[ssdp.ATTR_SSDP_UDN]
 
         # Abort if already configured, but update the last-known location
-        await self.async_set_unique_id(udn)
+        await self.async_set_unique_id(self._udn)
         self._abort_if_unique_id_configured(
-            updates={CONF_URL: location}, reload_on_update=False
+            updates={CONF_URL: self._location}, reload_on_update=False
         )
 
-        # If the device needs migration because it wasn't turned on when HA
-        # started, silently migrate it now.
-        import_data = get_domain_data(self.hass).unmigrated_config.get(location)
-        if import_data is not None:
-            return await self.async_step_import(import_data)
+        # Abort if a migration flow for the device's location is in progress
+        for progress in self._async_in_progress(include_uninitialized=True):
+            if progress["context"].get("unique_id") == self._location:
+                LOGGER.debug(
+                    "Aborting SSDP setup because migration for %s is in progress",
+                    self._location,
+                )
+                raise data_entry_flow.AbortFlow("already_in_progress")
```

Comment:
We're inside a step method so we can use `self.async_abort`.

Reply:
Won't `self.async_set_unique_id()` already abort the flow before this can be reached?

Reply:
In the import flow we set the unique_id to the location while here we set the unique_id to the udn, so the `async_set_unique_id` call won't abort automatically here.

Reply:
Indeed. The UDN isn't known early in the import flow so the location is used instead to identify and deconflict running import flows.